src/ui/components/Events/Event.tsx

```diff
@@ -31,15 +162,142 @@ export const getEventLabel = (event: ReplayEvent) => {
+  })()
+`;
+
+/*
+Jump to the function location that ran for a given info sidebar "Click" event list item.
+
+This requires stringing together a series of assumptions and special cases:
+
+- The info sidebar "Click" events come from `Session.findMouseEvents`.
+  These events are recorded in our browser forks _before_ any actual JS code runs.
+- However, we can assume that a _real_ `"click"` event occurs shortly thereafter.
+- We can find that "click" event based on a timeboxed search, with the initial event time
+  as the starting point.
+- We can use the "click" event's stack frame to know the location of the JS event handler
+  that started running in response to the event. _However_, React attaches noop handlers,
+  and implements its own event listener lookups at the top level (delegation).
+- Fortunately, React 16/17/18 add a secret expando property to DOM nodes, containing 
+  the actual props that the user rendered for that DOM node, such as `onClick`.
+- All "click" events will have a `MouseEvent` as one of their arguments
+- Once we find the `MouseEvent`, we can look at `event.target` to find the clicked node
+- But, the _target_ may not have had the handler due to delegation - the user-provided
+  React `onClick` prop may have been on an ancestor DOM node instead.
+- So, we can walk up the parent node chain and find the first node that has a React
+  `onClick` prop attached to it, if any, and return that
+- In order to optimize the API calls and network traffic, that parent node traversal
+  is done via a single JS evaluation, which returns `{target, onClick?}`.
+- If we found a React `onClick` in the chain, jump to that location. Otherwise,
+  jump to the location of the plain JS event handler in the stack frame.
+
+We _could_ do more analysis and find the nearest time where the first breakable location
+inside that function is running, then seek to that point in time, but skipping for now.
+*/
+function jumpToClickEventFunctionLocation(
+  event: ReplayMouseEvent,
+  onSeek: (point: ExecutionPoint, time: number) => void
+): UIThunkAction {
+  return async (dispatch, getState, { ThreadFront, replayClient }) => {
+    const { point: executionPoint, time } = event;
+    try {
+      // The sidebar event time/point is a fraction earlier than any
+      // actual JS that executed in response. Find the next click event
+      // within a small time window
+      const nextClickEvent = await getNextClickEventAsync(replayClient, executionPoint, time);
+
+      if (!nextClickEvent) {
+        return;
+      }
+
+      const pauseId = await getPauseIdAsync(
+        replayClient,
+        nextClickEvent.point,
+        nextClickEvent.time
+      );
+
+      // If we did have a click event, timewarp to that click's point
+      onSeek(nextClickEvent.point, nextClickEvent.time);
+
+      const sourceLocation = await getClickEventListenerLocationAsync(
+        ThreadFront,
+        replayClient,
+        getState,
+        pauseId
+      );
```

Comment:
Should we get this first, and only _seek_ if we can find a location too? Otherwise the timeline would jump to...nothing?

Reply:
Not sure I understand this question.

Previously, clicking any of the info sidebar list items unconditionally did a `seek()` to that execution point, period.  As I've found, that time is always offset from the actual browser event listener handling.  So, it gets you the general visuals of what's in the page and where the click event occurred, but not the exact line of code execution point.

Now, it does the seek to that initial execution point, but then _tries_ to refine it to the time of the click event listener if possible.

Reply:
That's fair!